exporter/elasticsearchexporter/bulkindexer.go

```diff
@@ -124,8 +126,17 @@ type syncBulkIndexerSession struct {
 }
 
 // Add adds an item to the sync bulk indexer session.
-func (s *syncBulkIndexerSession) Add(_ context.Context, index string, document io.WriterTo, dynamicTemplates map[string]string) error {
-	return s.bi.Add(docappender.BulkIndexerItem{Index: index, Body: document, DynamicTemplates: dynamicTemplates})
+func (s *syncBulkIndexerSession) Add(ctx context.Context, index string, document io.WriterTo, dynamicTemplates map[string]string) error {
+	err := s.bi.Add(docappender.BulkIndexerItem{Index: index, Body: document, DynamicTemplates: dynamicTemplates})
```

Comment:
In future, we can look into having a size/len method for `BulkIndexerItem`. This will allow us to enforce better soft limit on the byte size:

1. Always add to the bulk indexer if it is empty.
2. For non-empty, consider byte size as a hard limit.

Reply:
Do you mind clarifying the proposed size/len method? Is it the size of the item, or the size of buffer? If it is the size of the item, it requires the caller to estimate the size of item. If it is the size of the buffer, the fact that it is a field under BulkIndexerItem looks strange.

Reply:
>  Is it the size of the item, or the size of buffer?

Size the item will take if added to the buffer (so metadata + encoded size of the item).

Reply:
> requires the caller to estimate the size of item

Due to the `io.WriterTo` interface? If that then I am not sure how much value that interface provides as in most cases in which docappender will be used, the buffer would be in memory and the size known. We could in theory have a custom interface that implements `io#WriterTo` and provide another method for size.

Reply:
Bear in mind that it may create a bit of duplication of logic as the caller (es exporter) has to know the number of bytes to encode such an entry, and it will need to know how docappender encodes the bulk request action line, which depends on index and dynamic templates, plus a newline char, and the actual document body, and potentially a newline char after that. Not to mention the bulk request action line may change on docappender updates. Sounds like unnecessary coupling.

Reply:
> The size method is implemented by docappender

Yes, exactly. It will be in the docappender logic.